src/main/java/jsr223/kubernetes/KubernetesScriptEngine.java

```diff
@@ -133,13 +129,27 @@ else if (k8sDeleteOnly) {
         return resultValue;
     }
 
+    private void initializeEngine() {
+        bindings.addBindingsAsEnvironmentVariables(context);
+        setScriptEngineBehaviorFromEnv();
+    }
+
     /**********************************************/
     /* Kubernetes script engine auxiliary methods */
     /**********************************************/
 
     private void writeKubernetesManifestFile(String k8s_manifest) {
         // Write k8s manifest to file
+        Map<String, String> environment = bindings.getEnvironment();
         try {
+            for (Map.Entry<String, String> variable : environment.entrySet()) {
+                if (k8s_manifest.contains("${" + variable.getKey() + "}")) {
+                    k8s_manifest = org.apache.commons.lang3.StringUtils.replace(k8s_manifest,
+                                                                                "${" + variable.getKey() + "}",
```

Comment:
you can eventually copy/paste, adapt and use the method VariableSubstitutor.replaceRecursively().

This allows recursive replacements such as ${variables_PARAM_${variables_PA_TASK_ITERATION}}